sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayData.scala

```diff
@@ -34,6 +36,32 @@ object ArrayData {
     case a: Array[Double] => UnsafeArrayData.fromPrimitiveArray(a)
     case other => new GenericArrayData(other)
   }
+
+
+  /**
+   * Allocate [[UnsafeArrayData]] or [[GenericArrayData]] based on given parameters.
+   *
+   * @param elementSize a size of an element in bytes
+   * @param numElements the number of elements the array should contain
+   * @param isPrimitiveType whether the type of an element is primitive type
+   * @param additionalErrorMessage string to include in the error message
+   */
+  def allocateArrayData(
+      elementSize: Int,
+      numElements : Long,
+      isPrimitiveType: Boolean,
+      additionalErrorMessage: String) : ArrayData = {
+    if (isPrimitiveType && !UnsafeArrayData.shouldUseGenericArrayData(elementSize, numElements)) {
+      UnsafeArrayData.forPrimitiveArray(Platform.BYTE_ARRAY_OFFSET, numElements.toInt, elementSize)
```

Comment:
Do we need `Platform.BYTE_ARRAY_OFFSET` here? Seems like the method doesn't use the `offset` argument?

Reply:
Good catch